File: api/food-search.py
import os
import json
import urllib.request
import urllib.parse
from http.server import BaseHTTPRequestHandler

USDA_API_KEY = os.environ.get("USDA_API_KEY", "DEMO_KEY")

NUTRIENT_IDS = {
    "potassium":  1092,
    "phosphorus": 1091,
    "sodium":     1093,
    "calories":   1008,
    "protein":    1003,
    "water":      1051,
}


class handler(BaseHTTPRequestHandler):
    def do_GET(self):
        parsed = urllib.parse.urlparse(self.path)
        params = urllib.parse.parse_qs(parsed.query)
        query = params.get("q", [""])[0].strip()

        if not query:
            self._json([], 200)
            return

        usda_params = urllib.parse.urlencode({
            "query": query,
            "api_key": USDA_API_KEY,
            "pageSize": 8,
            "dataType": ["Foundation", "SR Legacy"],
        }, doseq=True)

        try:
            url = f"https://api.nal.usda.gov/fdc/v1/foods/search?{usda_params}"
            with urllib.request.urlopen(url, timeout=8) as resp:
                data = json.loads(resp.read())
        except Exception as e:
            self._json({"error": str(e)}, 502)
            return

        results = []
        for food in data.get("foods", []):
            nutrients = {
                n["nutrientId"]: round(n.get("value") or 0, 1)
                for n in food.get("foodNutrients", [])
            }
            results.append({
                "fdcId":       food.get("fdcId"),
                "description": food.get("description", "Unknown"),
                "potassium":   nutrients.get(NUTRIENT_IDS["potassium"],  0),
                "phosphorus":  nutrients.get(NUTRIENT_IDS["phosphorus"], 0),
                "sodium":      nutrients.get(NUTRIENT_IDS["sodium"],     0),
                "calories":    nutrients.get(NUTRIENT_IDS["calories"],   0),
                "protein":     nutrients.get(NUTRIENT_IDS["protein"],    0),
                "water":       nutrients.get(NUTRIENT_IDS["water"],      0),
            })

        self._json(results, 200)

    def _json(self, data, status):
        body = json.dumps(data).encode()
        self.send_response(status)
        self.send_header("Content-Type", "application/json")
        self.send_header("Access-Control-Allow-Origin", "*")
        self.send_header("Content-Length", str(len(body)))
        self.end_headers()
        self.wfile.write(body)

    def log_message(self, *args):
        pass

File: api/food-search.js
const NUTRIENT_IDS = {
  potassium:  1092,
  phosphorus: 1091,
  sodium:     1093,
  calories:   1008,
  protein:    1003,
  water:      1051,
};

export default async function handler(req, res) {
  res.setHeader("Access-Control-Allow-Origin", "*");

  const q = req.query.q?.trim();
  if (!q) return res.status(200).json([]);

  const USDA_KEY = process.env.USDA_API_KEY || "DEMO_KEY";
  const url = `https://api.nal.usda.gov/fdc/v1/foods/search?query=${encodeURIComponent(q)}&api_key=${USDA_KEY}&pageSize=8&dataType=Foundation&dataType=SR%20Legacy`;

  try {
    const r = await fetch(url, { signal: AbortSignal.timeout(8000) });
    const data = await r.json();

    const results = (data.foods || []).map(food => {
      const nutrients = Object.fromEntries(
        (food.foodNutrients || []).map(n => [n.nutrientId, +(n.value || 0).toFixed(1)])
      );
      return {
        fdcId:       food.fdcId,
        description: food.description,
        potassium:   nutrients[NUTRIENT_IDS.potassium]  || 0,
        phosphorus:  nutrients[NUTRIENT_IDS.phosphorus] || 0,
        sodium:      nutrients[NUTRIENT_IDS.sodium]     || 0,
        calories:    nutrients[NUTRIENT_IDS.calories]   || 0,
        protein:     nutrients[NUTRIENT_IDS.protein]    || 0,
        water:       nutrients[NUTRIENT_IDS.water]      || 0,
      };
    });

    return res.status(200).json(results);
  } catch (err) {
    return res.status(502).json({ error: err.message });
  }
}
